core/src/main/java/com/digitalpebble/stormcrawler/util/ConfUtils.java

```diff
@@ -14,73 +14,240 @@
+    }
+
+    /**
+     * Returns either a collection of objects or throws a RuntimeException. If there is only one
+     * value, it will return a SingletonList.
+     */
+    @NotNull
+    public static Collection<Object> loadCollection(
+            @NotNull Map<String, Object> conf, @NotNull String key) {
+        Collection<Object> collection = loadCollectionOrNull(conf, key);
+        if (collection == null) {
+            throw new RuntimeException(
+                    String.format("There is no Collection value for the key '%s'.", key));
+        }
+        return collection;
+    }
+
+    /**
+     * Returns either a collection ob objects or null. If there is only one value, it will return a
+     * SingletonList.
+     */
+    @Nullable
+    public static Collection<Object> loadCollectionOrNull(
+            @NotNull Map<String, Object> conf, @NotNull String key) {
+        Object obj = conf.get(key);
+        if (obj == null) {
+            return null;
+        }
+        if (obj instanceof Collection) {
+            //noinspection unchecked
+            return (Collection<Object>) obj;
+        } else {
+            return Collections.singletonList(obj);
+        }
+    }
+
+    @Nullable
+    public static Map<String, Object> getSubConfig(
+            @NotNull Map<String, Object> conf, @NotNull String key) {
+        Object obj = conf.get(key);
+        if (obj == null) {
+            return null;
+        }
+        //noinspection unchecked
+        return (Map<String, Object>) obj;
+    }
+
+    /**
+     * Returns either a Map or a reference to {@link Collections#EMPTY_MAP}. If {@code key} is not
+     * in {@code conf} it returns null.
+     */
+    @Nullable
+    public static Map<String, Object> loadSubConfigOrNull(
+            @NotNull Map<String, Object> conf, @NotNull String key) {
+        Object obj = conf.get(key);
+        if (obj == null) {
+            return null;
+        }
+        if (obj instanceof Map) {
+            //noinspection unchecked
+            return (Map<String, Object>) obj;
+        } else {
+            return Collections.emptyMap();
+        }
+    }
+
+    /**
+     * Returns either a Map or a reference to {@link Collections#EMPTY_MAP}. If {@code key} is not
+     * in {@code conf} it trows a RuntimeCollection.
+     */
+    @NotNull
+    public static Map<String, Object> loadSubConfig(
+            @NotNull Map<String, Object> conf, @NotNull String key) {
+        Map<String, Object> subConfig = loadSubConfigOrNull(conf, key);
+        if (subConfig == null) {
+            throw new RuntimeException(String.format("There is no SubConfig at key '%s'.", key));
+        }
+        return subConfig;
+    }
+
     /**
      * Return one or more Strings regardless of whether they are represented as a single String or a
      * list in the config or an empty List if no value could be found for that key.
      */
-    public static List<String> loadListFromConf(String paramKey, Map stormConf) {
```

Comment:
Might break user code, maybe provide additional `public` method with original signature?

Reply:
yeah, i forgot to deprecate it. ^^'
I'll correct it when I get an approval of jnioche that he accepts the presented feature.